src/langchain_google_bigtable/document_loader.py

```diff
@@ -0,0 +1,265 @@
+        self.custom_decoding_func = custom_decoding_func
+
+
+SUPPORTED_DOC_ENCODING = (Encoding.UTF8, Encoding.UTF16, Encoding.ASCII)
+
+
+class BigtableLoader(BaseLoader):
+    """Load from the Google Cloud Platform `Bigtable`."""
+
+    def __init__(
+        self,
+        instance_id: str,
+        table_id: str,
+        row_set: Optional[bigtable.row_set.RowSet] = None,
+        filter: Optional[bigtable.row_filters.RowFilter] = None,
+        client: Optional[bigtable.Client] = None,
+        content_encoding: Encoding = Encoding.UTF8,
+        content_column_family: str = COLUMN_FAMILY,
+        content_column_name: str = CONTENT_COLUMN_NAME,
+        metadata_mappings: List[MetadataMapping] = [],
+    ):
+        """Initialize Bigtable document loader.
+
+        Args:
+            instance_id: The Bigtable instance to load data from.
+            table_id: The Bigtable table to load data from.
+            row_set: Optional. The row set to read data from.
+            filter: Optional. The filter to apply to the query to load data from bigtable.
+            client : Optional. The pre-created client to query bigtable.
+            content_encoding: Optional. The encoding in which to load the page content. Defaults to UTF-8. Allowed values are  UTF8, UTF16 and ASCII.
+            content_column_family: Optional. The column family in which the content is stored. Defaults to "langchain".
+            content_column_name: Optional. The column in which the content is stored. Defaults to "content".
+            metadata_mappings: Optional. The array of mappings that maps from Bigtable columns to keys on the metadata dictionary, including the encoding to use when mapping from Bigtable bytes to a python type.
+        """
+        self.row_set = row_set
+        self.filter = filter
+        self.client = (
+            (client or bigtable.Client(admin=True))
+            .instance(instance_id)
+            .table(table_id)
+        )
+        if content_encoding not in SUPPORTED_DOC_ENCODING:
+            raise ValueError(f"{content_encoding} not in {SUPPORTED_DOC_ENCODING}")
+        families = self.client.list_column_families()
+        for mapping in metadata_mappings:
+            if mapping.column_family not in families:
+                raise ValueError(
+                    f"column family '{mapping.column_family}' doesn't exist in table. Existing column families are {families.keys()}"
+                )
+        self.content_encoding = content_encoding
+        self.content_column_family = content_column_family
+        self.content_column_name = content_column_name
+        self.metadata_mappings = metadata_mappings
+
+    def load(self) -> List[Document]:
+        return list(self.lazy_load())
+
+    def lazy_load(
+        self,
+    ) -> Iterator[Document]:
+        """A lazy loader for Documents."""
+        rows = self.client.read_rows(row_set=self.row_set, filter_=self.filter)
+        for row in rows:
```

Comment:
does this handle pagination/streaming automatically?

Reply:
streaming yes, pagination no.
A user can use a filter or row_set to ask for a single page, if they want.
It is possible for us to add a "limit" parameter to only get N rows at a time. Should we do that?

Reply:
If it's streaming a few at time I think that's okay --just want to avoid loading everything at once.